src/types/server_urls/server_urls_bucket.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{constants::STREAMING_SERVER_URL, runtime::Env, types::profile::UID};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use url::Url;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq)]
+pub struct ServerUrlsBucket {
+    /// User ID
+    pub uid: UID,
+    /// HashMap where key is `Url`, value is `DateTime<Utc>`
+    pub items: HashMap<Url, DateTime<Utc>>,
```

Comment:
There's one issue with the hashmap - it doesn't keep the ordering.
If you need for the items' order to be kept you can check out e.g. indexmap crate.

- [ ] confirm whether or not we need to keep the order of items

Reply:
Done by comparing the modified(or added) date time